device/common/include/traccc/device/concepts/thread_id.hpp

```diff
@@ -19,7 +19,7 @@ namespace traccc::device::concepts {
  * @tparam T The thread identifier-like type.
  */
 template <typename T>
-concept thread_id1 = requires(T& i) {
+concept thread_id1 = requires(const T& i) {
```

Comment:
If we're making the `thread_id` objects templated on the dimensionality, why not the `thread_id` concept too? :wink:

Reply:
I thought about that. But decided that I like the non-templated concept better. :thinking:

We use very specific function names in the concept. In a higher dimensionality concept we would probably replace `X` in a couple of the function names with `Y`.

I think I'd rather just add `thread_id2` as a concept once we need it. (I imagine it would require the `thread_id1` concept, and add some more required functions.) That would still allow us to be pretty free with how we provide the implementation classes.